src/common/tracer.cc

```diff
@@ -55,6 +65,37 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    ENCODE_START(1, 1, bl);
+    encode_nohead(std::string_view((char*)span_ctx.trace_id().Id().data(), TraceId::kSize), bl);
+    encode_nohead(std::string_view((char*)span_ctx.span_id().Id().data(), SpanId::kSize), bl);
+    encode(span_ctx.trace_flags().flags(), bl);
+    ENCODE_FINISH(bl);
+  }
+}
+
+void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {
+  if (bl.get_remaining() == 0) {
+    return ;
+  }
+  using namespace opentelemetry;
+  using namespace trace;
+  const char* trace_id_ptr;
+  const char* span_id_ptr;
+  uint8_t flags;
+  DECODE_START(1, bl);
+  bl.get_ptr_and_advance(TraceId::kSize, &trace_id_ptr);
+  bl.get_ptr_and_advance(SpanId::kSize, &span_id_ptr);
+  decode(flags, bl);
+  DECODE_FINISH(bl);
+  nostd::span<uint8_t, TraceId::kSize> trace((uint8_t*)trace_id_ptr, TraceId::kSize);
```

Comment:
can we try to use: `nostd::span<char*, TraceId::kSize>`
so we don't need the c-style cast?

Reply:
no, 
but we can use the `ToLowerBase16` method (instead of `CopyBytesTo`), which returns char array that represents the traec_id. but the size of that array is `2 * TraceId::kSize`,  because they represent each unsigned char by  2 hex chars. and then we can convert it back to `uint8_t` by a function called `HexToBinary`  (opentelemetry function).

Reply:
well, this would be too inefficient...
maybe we can get an unsigned char pointer from the bufferlist?